Index: shareneat/src/main/java/gtu/codybuilders/shareneat/util/AuthUtil.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package gtu.codybuilders.shareneat.util;\r\n\r\nimport gtu.codybuilders.shareneat.model.UserPrincipal;\r\nimport org.springframework.security.core.Authentication;\r\nimport org.springframework.security.core.context.SecurityContextHolder;\r\nimport org.springframework.stereotype.Component;\r\n\r\n@Component\r\npublic class AuthUtil {\r\n\r\n    public static Long getUserId() {\r\n        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();\r\n\r\n        if (authentication != null && authentication.isAuthenticated()) {\r\n            Object principal = authentication.getPrincipal();\r\n\r\n            if (principal instanceof UserPrincipal) {\r\n                return ((UserPrincipal) principal).getUser().getUserId();\r\n            }\r\n        }\r\n\r\n        throw new IllegalStateException(\"No authenticated user found\");\r\n    }\r\n\r\n}\r\n
===================================================================
diff --git a/shareneat/src/main/java/gtu/codybuilders/shareneat/util/AuthUtil.java b/shareneat/src/main/java/gtu/codybuilders/shareneat/util/AuthUtil.java
--- a/shareneat/src/main/java/gtu/codybuilders/shareneat/util/AuthUtil.java	
+++ b/shareneat/src/main/java/gtu/codybuilders/shareneat/util/AuthUtil.java	
@@ -8,6 +8,10 @@
 @Component
 public class AuthUtil {
 
+    /**
+     * Retrieves the user ID of the currently authenticated user.
+     * @return the user ID of the authenticated user.
+     */
     public static Long getUserId() {
         Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
 
